one_fm/grd/doctype/mgrp/mgrp.py

```diff
@@ -5,26 +5,32 @@
 from frappe.model.document import Document
 import frappe
 from frappe import _
-from frappe.utils import get_url
+from frappe.utils import get_url, cint
 from one_fm.api.notification import create_notification_log
 # from frappe.utils import today
 from datetime import date
 from one_fm.hiring.utils import update_onboarding_doc
-
+from frappe.core.doctype.communication.email import make
 class MGRP(Document):
 	def validate(self):
-		self.children_table()
+		self.get_children_table()
 		self.set_grd_values()
 		self.set_status()
-		if not self.end_of_service_attachment:
+		self.set_progress()
+		if self.status == "Cancellation" and not self.end_of_service_attachment:
 			self.set_resignation_attachment()
 		if not self.date_of_application:
-			self.set_date_of_applicantion_value()
+			self.db_set('date_of_application',date.today())
```

Comment:
On validate you can just put `self.date_of_application = today()`, today can be imported from `frappe.utils`